### PR TITLE
HcContributionForm submit is disabled by default

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
HcContributionForm submit is disabled by default

ezgif com-video-to-gif (2)

### Issues
- fixes #https://github.com/Human-Connection/Human-Connection/issues/806

### CODE REVIEW

File: webapp/components/ContributionForm/ContributionForm.spec.js [hunk 1]
```diff
@@ -201,7 +220,12 @@ describe('ContributionForm.vue', () => {
             content: 'auf Deutsch geschrieben',
             language: 'de',
             image,
-            categories: [{ id: 'cat12', name: 'Democracy & Politics' }],
+            categories: [
+              {
+                id: 'cat12',
+                name: 'Democracy & Politics',
+              },
+            ],
```
Comment (roschaefer) [inline webapp/components/ContributionForm/ContributionForm.spec.js]: @ogerly try not to touch unrelated files. First, it makes revieweing your PR difficult. Second and more important, you will "own" the code. This will make it difficult to understand the code, because `git blame` will tell me that you are the author, but you just changed indentation.

To fix this: Run`git checkout origin/master -- path/to/unrelated/files`.

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 2]
```diff
@@ -169,6 +169,16 @@ export default {
     },
     updateEditorContent(value) {
       // this.form.content = value
+      this.disabled = true
+      if (
+        value.replace(
+          /<p>|<\/p>|<h3>|<\/h3>|<h4>|<\/h4>|<ul>|<\/ul>|<ol>|<\/ol>|<li>|<\/li>|<blockquote>|<\/blockquote>|<strong>|<\/strong>|<em>|<\/em>|<a.*>|<\/a>/gm,
```
Comment (roschaefer) [inline webapp/components/ContributionForm/ContributionForm.vue]: Please don't use a regex. If you want to strip HTML from a string, the [internet has the answer](https://stackoverflow.com/questions/5002111/how-to-strip-html-tags-from-string-in-javascript).. Regexes are horrendously hard to maintain. "If you found out you can solve your problem with a regex, you have a new problem."

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 3]
```diff
@@ -169,6 +169,16 @@ export default {
...
+          '',
+        ).length > 3
```
Comment (roschaefer) [inline webapp/components/ContributionForm/ContributionForm.vue]: Its' quite obscure what you do here. I guess you want to avoid validation errors in the backend? Do we even have a validation for the length of a `post.content`? @ogerly could you double-check?

Anyways, ideally, you would explain what you do here with a test case :+1:

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 2 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: to count the absolute characters, a lenght function is not enough. because the editor just inserts these html tags. so i only count the real characters that a user writes. nothing will be deleted or added from the original text. 

since I hold myself back from inserting npm packete. 

or I don't know if this already exists. 

since i love regular expressions that was my solution :)

Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: no the number of characters in a content is not limited . i would suggest 2000 characters. maybe 4000. you can say badly .. some like to write whole romanes. maybe there should be a BLOG corner for every user for something like that. 
i think 2000 characters is ok. those are 2 A4 pages

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 3 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: no the number of characters in a content is not limited . i would suggest 2000 characters. maybe 4000. you can say badly .. some like to write whole romanes. maybe there should be a BLOG corner for every user for something like that. 
i think 2000 characters is ok. those are 2 A4 pages

File: webapp/components/ContributionForm/ContributionForm.spec.js (hunk 1 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.spec.js]: yes, i'm sorry. 
i have a lint problem in my editor. 
when i edit a page that i later restore to its original state. there is still a change in the git status due to the lint. 
should i just turn it off in my editor ? 
Bildschirmfoto vom 2019-07-10 06-51-51

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 3 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: no the number of characters in a content is not limited . i would suggest 2000 characters. maybe 4000. you can say badly .. some like to write whole romanes. maybe there should be a BLOG corner for every user for something like that. 
i think 2000 characters is ok. those are 2 A4 pages

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 2 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: replace("<[^>]*>", "")  is the short code ;)

Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: the short code is: .replace(/<\/?[^>]+(>|$)  ;)

**Comment by ogerly** [comment]
maximum number of characters in content without html tags is 2000, characters are counted and number is displayed. regex is strongly shortened.

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 3 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: FireShot Capture 174 - Human Connection - Human Connection - localhost

Comment (roschaefer) [reply webapp/components/ContributionForm/ContributionForm.vue]: But I think we're not talking about maximum but *minimum* characters, right? Or how do you explain what your code does?

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 4]
```diff
@@ -10,13 +10,15 @@
           />
         </hc-teaser-image>
         <ds-input model="title" class="post-title" placeholder="Title" name="title" autofocus />
+        <small style="width:100%;position:relative;left:90%">{{ form.title.length }}/64</small>
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]: I prefer to avoid inline styling @ogerly, not sure how others fell about it

**Comment by mattwr18** [comment]
@roschaefer, this is still waiting your approval before getting merged...

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 4 above)
Comment (Tirokk) [reply webapp/components/ContributionForm/ContributionForm.vue]: Would be nice(!) to make a class out of it … @ogerly 
But not necessarily …
Inline styling has the positive side of better analysability, if the same style is not used elsewhere.
Classes (can) have descriptive names telling us what this is about.

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 5]
```diff
@@ -10,13 +10,15 @@
...
         <no-ssr>
           <hc-editor
             :users="users"
             :hashtags="hashtags"
             :value="form.content"
             @input="updateEditorContent"
           />
+          <small style="width:100%;position:relative;left:90%">{{ form.contentLength }}/2000</small>
```
Comment (Tirokk) [inline webapp/components/ContributionForm/ContributionForm.vue]: I would like more to have this not a hard code max length and been a speaking and centralised changeable const …

Ah, you have defined it here: `this.formSchema.content.max`

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 6]
```diff
@@ -94,13 +97,14 @@ export default {
       },
       formSchema: {
         title: { required: true, min: 3, max: 64 },
-        content: { required: true, min: 3 },
+        content: { required: true, min: 3, max: 2000 },
```
Comment (Tirokk) [inline webapp/components/ContributionForm/ContributionForm.vue]: Same same 👆🏼

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 7]
```diff
@@ -177,6 +181,14 @@ export default {
     updateEditorContent(value) {
       // this.form.content = value
       this.$refs.contributionForm.update('content', value)
+
+      this.disabled = true
+      this.n = value.replace(/<\/?[^>]+(>|$)/gm, '').length
+      this.form.contentLength = this.n
+
+      if (this.n > this.formSchema.content.min && this.n < this.formSchema.content.max) {
+        this.disabled = false
```
Comment (Tirokk) [inline webapp/components/ContributionForm/ContributionForm.vue]: `this.n` should be a local `let` variable and not be global to this component …

File: webapp/layouts/default.vue [hunk 8]
```diff
@@ -228,7 +228,7 @@ export default {
 
 .main-container {
   padding-top: 6rem;
-  padding-botton: 5rem;
+  padding-bottom: 5rem;
```
Comment (Tirokk) [inline webapp/layouts/default.vue]: TOP !!! 👍🏼

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 9]
```diff
@@ -177,6 +181,14 @@ export default {
     updateEditorContent(value) {
       // this.form.content = value
       this.$refs.contributionForm.update('content', value)
+
+      this.disabled = true
+      this.n = value.replace(/<\/?[^>]+(>|$)/gm, '').length
+      this.form.contentLength = this.n
+
+      if (this.n > this.formSchema.content.min && this.n < this.formSchema.content.max) {
```
Comment (Tirokk) [inline webapp/components/ContributionForm/ContributionForm.vue]: Here you use `this.formSchema.content.max` already! 👋🏼

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 10]
```diff
@@ -177,6 +181,14 @@ export default {
     updateEditorContent(value) {
       // this.form.content = value
```
Comment (Tirokk) [inline webapp/components/ContributionForm/ContributionForm.vue]: Remove this comment?

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 7 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: amended
